components/nvs/include/nvs.hpp

```diff
@@ -70,6 +70,225 @@ const NvsErrCategory theNvsErrCategory{};
 } // namespace
 
 namespace espp {
+class NVSHandle : public BaseComponent {
+public:
+  /// @brief Construct a new NVSHandle object
+  /// @param[in] ns_name Namespace for NVS
+  /// @param[out] ec Saves a std::error_code representing success or failure
+  /// @details Create an NVSHandle object for the key-value pairs in the ns_name namespace
+  explicit NVSHandle(const char *ns_name, std::error_code &ec)
+      : BaseComponent("NVSHandle", espp::Logger::Verbosity::WARN) {
+    esp_err_t err;
+    ns_name = ns_name;
+    handle = nvs::open_nvs_handle(ns_name, NVS_READWRITE, &err);
+    if (err != ESP_OK) {
+      logger_.error("Error {} opening NVS handle for namespace '{}'!", esp_err_to_name(err),
+                    ns_name);
+      ec = make_error_code(NvsErrc::Open_NVS_Handle_Failed);
+    }
+  }
+
+  /// @brief Reads a variable from the NVS
+  /// @param[in] key NVS Key of the variable to read
+  /// @param[in] value Variable to read
+  /// @param[out] ec Saves a std::error_code representing success or failure
+  /// @details Read the key/variable pair without committing
```

Comment:
1. reads don't require / involve committing
2. Probably best to just say read the value of of `key` into `variable`, if `key` exists.